src/ansys/speos/core/sensor.py

```diff
@@ -3742,3 +3742,617 @@ def set_geometries(self, geometries: [List[GeoRef]]) -> Sensor3DIrradiance:
+            LayerTypeFace property instance
+        """
+        if self._layer_type is None and self._sensor_instance.intensity_properties.HasField(
+            "layer_type_face"
+        ):
+            # Happens in case of project created via load of speos file
+            self._layer_type = BaseSensor.LayerTypeFace(
+                layer_type_face=self._sensor_instance.intensity_properties.layer_type_face,
+                default_values=False,
+                stable_ctr=True,
+            )
+        elif not isinstance(self._layer_type, BaseSensor.LayerTypeFace):
+            # if the _layer_type is not LayerTypeFace then we create a new type.
+            self._layer_type = BaseSensor.LayerTypeFace(
+                layer_type_face=self._sensor_instance.intensity_properties.layer_type_face,
+                stable_ctr=True,
+            )
+        elif (
+            self._layer_type._layer_type_face
+            is not self._sensor_instance.intensity_properties.layer_type_face
+        ):
+            # Happens in case of feature reset (to be sure to always modify correct data)
+            self._layer_type._layer_type_face = (
+                self._sensor_instance.intensity_properties.layer_type_face
+            )
+        return self._layer_type
+
+    def set_layer_type_sequence(self) -> BaseSensor.LayerTypeSequence:
+        """Define layer separation as by sequence.
+
+        Returns
+        -------
+        ansys.speos.core.sensor.BaseSensor.LayerTypeSequence
+            LayerTypeSequence property instance
+        """
+        if self._layer_type is None and self._sensor_instance.intensity_properties.HasField(
+            "layer_type_sequence"
+        ):
+            # Happens in case of project created via load of speos file
+            self._layer_type = BaseSensor.LayerTypeSequence(
+                layer_type_sequence=self._sensor_instance.intensity_properties.layer_type_sequence,
+                default_values=False,
+                stable_ctr=True,
+            )
+        elif not isinstance(self._layer_type, BaseSensor.LayerTypeSequence):
+            # if the _layer_type is not LayerTypeSequence then we create a new type.
+            self._layer_type = BaseSensor.LayerTypeSequence(
+                layer_type_sequence=self._sensor_instance.intensity_properties.layer_type_sequence,
+                stable_ctr=True,
+            )
+        elif (
+            self._layer_type._layer_type_sequence
+            is not self._sensor_instance.intensity_properties.layer_type_sequence
+        ):
+            # Happens in case of feature reset (to be sure to always modify correct data)
+            self._layer_type._layer_type_sequence = (
+                self._sensor_instance.intensity_properties.layer_type_sequence
+            )
+        return self._layer_type
+
+    def set_axis_system(self, axis_system: Optional[List[float]] = None) -> SensorXMPIntensity:
```

Comment:
this can be property & setter method, for sure.

Reply:
yeah just copied that^^ will refactor it